core/aws-core/src/test/java/software/amazon/awssdk/awscore/checksum/AwsSignerWithChecksumTest.java

```diff
@@ -44,7 +44,8 @@ public void signingWithChecksumWithSha256ShouldHaveChecksumInHeaders() throws Ex
         SdkHttpFullRequest.Builder request = generateBasicRequest("abc");
         ExecutionAttributes executionAttributes = getExecutionAttributes(SHA_256_HEADER);
         SdkHttpFullRequest signed = signer.sign(request.build(), executionAttributes);
-        final Optional<String> checksumHeader = signed.firstMatchingHeader("header-sha256");
+        // though the headerName is given, the signer must use the 'x-amz-checksum-<algorithm>' header
+        Optional<String> checksumHeader = signed.firstMatchingHeader("x-amz-checksum-sha256");
```

Comment:
If we agree that this behavior change is ok, then along with the above test update, we should update the tests below to additionally assert isNotPresent on the x-amz-checksum-<algorithm>. And I noticed signingWithNoChecksumHeaderAlgorithmShouldNotAddChecksumInHeaders and signingWithNoChecksumShouldNotHaveChecksumInHeaders are exactly same, seems like that's one test signingWithNoChecksumSpecsShouldNotHaveChecksumInHeaders.

Reply:
hmm, regarding the isNotPresent test, since there is no specific algorithm, there isn't a specific header name to check absence of. maybe we can check there is no header that starts with "x-amz-checksum"

Reply:
hmm, is my understanding correct that what this test is also telling me is that just a write, read on this attribute will change the value, without there being an actual SelectedAuthScheme in the attributes?

i.e., this will fail?
```
ExecutionAttributes.putAttribute(RESOLVED_CHECKSUM_SPECS, in);
out = ExecutionAttributes.getAttribute(RESOLVED_CHECKSUM_SPECS);
assertThat(out).isSameAs(in);
```

I'm wondering even if this test (i.e., the Signer behavior) is ok, should the use of the right header be responsibility of the Signer v/s mapping logic happening in ExecutionAttribute classes.

Though, I'm not sure if it will matter anywhere or whether a fix to avoid this would be worth it.

Reply:
I've updated the changes to account for compatibility. The mapping is lossless now, so this shouldn't technically be an issue anymore.